src/xcoder.rs

```diff
@@ -0,0 +1,247 @@
+use std::io::Cursor;
+use std::sync::Arc;
+
+use ::image::codecs::png::{PngDecoder, PngEncoder};
+use ::image::{ColorType, DynamicImage, ImageDecoder, Rgba, RgbaImage};
+use tokio::runtime::Runtime;
+use tokio::sync::{mpsc, oneshot};
+use tokio::task::JoinHandle;
+
+use crate::chunk::{Chunk, ChunkImage};
+use crate::Error;
+
+pub struct XcoderChannels {
+   // Decoder
+   pub decoded_chunks_rx: mpsc::UnboundedReceiver<((i32, i32), RgbaImage)>,
+
+   // Encoder
+   pub encoded_chunks_rx: mpsc::UnboundedReceiver<((i32, i32), ChunkImage)>,
```

Comment:
Any reason to have these comments here? I don't think they add much value on top of the field names already.